src/utilities/urlAssembly.ts

```diff
@@ -1,10 +1,11 @@
 import { Param, Resolved, Route } from '@/types'
 import { setParamValuesOnUrl } from '@/utilities/paramsFinder'
 
-export function assembleUrl(route: Resolved<Route>, values: Record<string, unknown> = {}): string {
+export function assembleUrl(route: Resolved<Route>, values: Record<string, unknown[]> = {}): string {
   const params = Object.entries<Param[]>(route.params)
+  const pathWithQuery = route.query.length ? `${route.path}?${route.query}` : route.path
```

Comment:
I considered moving this out into it's own utility

Reply:
This whole utility is pretty simple. So this ternary doesn't bother me.